com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/Shaders/IndirectDiffuse/RaytracingIndirectDiffuse.compute

```diff
@@ -128,109 +132,323 @@ Texture2DArray<float> _BlueNoiseTexture;
 
 // Output Textures for the spatial filtering
 RW_TEXTURE2D_X(float4, _UpscaledIndirectDiffuseTextureRW);
-int _SpatialFilterRadius;
+
+#define NORMAL_REJECTION_THRESHOLD 0.5
+
+struct NeighborTapData
+{
+    float3 lighting;
+    float linearDepth;
+    float3 normalWS;
+};
+
+float EvaluateNeighborWeight(in NeighborTapData neighborData, float3 normalWS, float linearDepth)
+{
+    // Initially the weight of this pixel is evaluate using the depth
+    float weight = 1.0f;
+
+    // If the candidate pixel is a background pixel, we cannot use it
+    if(neighborData.linearDepth == 1.0)
+        weight = 0.0f;
+
+    // If the normals of both pixels are too different, we cannot use it
+    if(dot(neighborData.normalWS, normalWS) < NORMAL_REJECTION_THRESHOLD)
+        weight *= 0.1f;
```

Comment:
Would it make sense to soft reject this? Lerping towards 0.1 (or close to) multiplier as we approach the threshold?

Reply:
I guess we can have a transition region. But the idea here is to not use these unless there are not better samples